pkg/deploy/discover/script_discover.go

```diff
@@ -59,51 +60,72 @@ func (sd *ScriptDiscoverer) GetTaskConfigs(ctx context.Context, file string) ([]
 		return nil, err
 	}
 
-	r, err := runtime.Lookup(file, task.Kind)
+	pathMetadata, err := taskPathMetadata(file, task.Kind)
 	if err != nil {
-		return nil, errors.Wrapf(err, "cannot determine how to deploy %q - check your CLI is up to date", file)
+		return nil, err
+	}
+	def.SetBuildConfig("entrypoint", pathMetadata.Entrypoint)
+	if err := def.SetWorkdir(pathMetadata.TaskRoot, pathMetadata.Workdir); err != nil {
+		return nil, err
+	}
+
+	return []TaskConfig{
+		{
+			TaskID:         task.ID,
+			TaskRoot:       pathMetadata.TaskRoot,
+			TaskEntrypoint: pathMetadata.AbsFile,
+			Def:            def,
+			Source:         sd.ConfigSource(),
+		},
+	}, nil
+}
+
+func (sd *ScriptDiscoverer) ConfigSource() ConfigSource {
+	return ConfigSourceScript
+}
+
+type TaskPathMetadata struct {
+	AbsFile    string
+	Entrypoint string
+	TaskRoot   string
+	Workdir    string
```

Comment:
Minor, these names are more clear IMO:

```suggestion
	AbsEntrypoint    string
	RelEntrypoint string
	RootDir   string
	WorkDir    string
```